### PR TITLE
profiler: Expose missing profile stacks count as a prometheus metric

### PR DESCRIPTION
**Pull request by Sylfrena**
Signed-off-by: Sumera Priyadarsini <sylphrenadin@gmail.com>

### CODE REVIEW

File: cmd/parca-agent/main.go [hunk 1]
```diff
@@ -367,7 +367,7 @@ func main() {
 		ctx, cancel := context.WithCancel(ctx)
 		g.Add(func() error {
 			level.Debug(logger).Log("msg", "starting target manager")
-			return tm.Run(ctx, m.SyncCh())
+			return tm.Run(ctx, m.SyncCh(), reg)
```
Comment (kakkoyun) [inline cmd/parca-agent/main.go:370]: Better to move passing the registry to the constructor.

**Review by kakkoyun — COMMENTED** [review]
Looking good. 
I have pointed out a couple of minor things.

File: cmd/parca-agent/main.go [hunk 2]
```diff
@@ -158,7 +158,7 @@ func main() {
 	}
 
 	externalLabels := getExternalLabels(flags.ExternalLabel, flags.Node)
-	tm := target.NewManager(logger, externalLabels, ksymCache, listener, dc, flags.ProfilingDuration, flags.TempDir)
+	tm := target.NewManager(logger, reg, externalLabels, ksymCache, listener, dc, flags.ProfilingDuration, flags.TempDir)
```
Comment (kakkoyun) [inline cmd/parca-agent/main.go:161]: 👍

File: pkg/profiler/profiler.go [hunk 3]
```diff
@@ -94,6 +100,12 @@ func NewCgroupProfiler(
 			tmp,
 		),
 		mtx: &sync.RWMutex{},
+		missingStacks: promauto.With(reg).NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "parca_agent_profiler_missing_stacks",
+				Help: "Number of missing profile stacks",
+			},
+			[]string{"stacktype"}),
```
Comment (kakkoyun) [inline pkg/profiler/profiler.go:108]:
```suggestion
			[]string{"type"}),
```

File: pkg/profiler/profiler.go [hunk 4]
```diff
@@ -300,7 +312,7 @@ func (p *CgroupProfiler) profileLoop(ctx context.Context, now time.Time, counts,
 
 		stackBytes, err := stackTraces.GetValue(unsafe.Pointer(&userStackID))
 		if err != nil {
-			//profile.MissingStacks++
+			p.missingStacks.WithLabelValues("userstack").Inc()
```
Comment (kakkoyun) [inline pkg/profiler/profiler.go:315]:
```suggestion
			p.missingStacks.WithLabelValues("user").Inc()
```

File: pkg/profiler/profiler.go [hunk 5]
```diff
@@ -314,7 +326,7 @@ func (p *CgroupProfiler) profileLoop(ctx context.Context, now time.Time, counts,
 		if kernelStackID >= 0 {
 			stackBytes, err = stackTraces.GetValue(unsafe.Pointer(&kernelStackID))
 			if err != nil {
-				//profile.MissingStacks++
+				p.missingStacks.WithLabelValues("kernelstack").Inc()
```
Comment (kakkoyun) [inline pkg/profiler/profiler.go:329]:
```suggestion
				p.missingStacks.WithLabelValues("kernel").Inc()
```

File: pkg/profiler/profiler.go [hunk 6]
```diff
@@ -94,6 +100,12 @@ func NewCgroupProfiler(
 			tmp,
 		),
 		mtx: &sync.RWMutex{},
+		missingStacks: promauto.With(reg).NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "parca_agent_profiler_missing_stacks",
```
Comment (kakkoyun) [inline pkg/profiler/profiler.go:105]:
```suggestion
				Name: "parca_agent_profiler_missing_stacks_total",
```

FYI https://prometheus.io/docs/practices/naming/

**Review by kakkoyun — CHANGES_REQUESTED** [review]
Looking great!

Just pointed out some name things.

File: pkg/profiler/profiler.go (hunk 6 above)
Comment (kakkoyun) [reply pkg/profiler/profiler.go:105]: e. g. https://github.com/parca-dev/parca/blob/6e21eb3fb2b1d8e330545b4cdb42e1a9ff7d2005/pkg/metastore/cache.go#L65-L121

**Review by kakkoyun — APPROVED** [review]

**Comment by metalmatze** [comment]
:rocket:

**Pull request merged**